From 9d394e80286f52803ffc33a8992094414c89144b Mon Sep 17 00:00:00 2001
From: allen yan <yanwei@marvell.com>
Date: Thu, 18 May 2017 17:16:02 +0800
Subject: [PATCH 1002/1345] fix: pm: uart: a3700: remove save and restore uart
 TSH register

commit  e9634ed07f0bb404c3d4e6bdc7cb19d90d2dc8cc from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

There is no need to save and restore uart Transmitter Holding
register. It may cause some print to console.
This patch remove save and restore uart TSH register.

Change-Id: Ia3f025d3a3e7f7c6fd43e0f074c11991ade78aeb
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39631
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/mvebu-uart.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/tty/serial/mvebu-uart.c b/drivers/tty/serial/mvebu-uart.c
index 4628690..e094291 100644
--- a/drivers/tty/serial/mvebu-uart.c
+++ b/drivers/tty/serial/mvebu-uart.c
@@ -1017,7 +1017,6 @@ static int mvebu_uart_reg_save(struct mvebu_uart_data *data)
 	if (data->reg_type == REG_UART_A3700_EXT)
 		data->pm_reg_value.uart_ctrl2 = readl(data->port->membase + REG_CTRL2(data));
 
-	data->pm_reg_value.uart_tsh = readl(data->port->membase + REG_TSH(data));
 	data->pm_reg_value.uart_brdv = readl(data->port->membase + REG_BRDV(data));
 	data->pm_reg_value.uart_over_sample = readl(data->port->membase + REG_OSAMP(data));
 	if (!IS_ERR_OR_NULL(data->intr.uart_int_base))
@@ -1034,7 +1033,6 @@ static int mvebu_uart_reg_restore(struct mvebu_uart_data *data)
 	if (data->reg_type == REG_UART_A3700_EXT)
 		writel(data->pm_reg_value.uart_ctrl2, data->port->membase + REG_CTRL2(data));
 
-	writel(data->pm_reg_value.uart_tsh, data->port->membase + REG_TSH(data));
 	writel(data->pm_reg_value.uart_brdv, data->port->membase + REG_BRDV(data));
 	writel(data->pm_reg_value.uart_over_sample, data->port->membase + REG_OSAMP(data));
 	if (!IS_ERR_OR_NULL(data->intr.uart_int_base))
-- 
1.7.9.5

